app/controllers/books_controller.rb

```diff
@@ -0,0 +1,52 @@
+class BooksController < ApplicationController
+  before_action :set_book, only: %i[show update destroy]
```

Comment:
We don't user before_actions in controllers :confused: (except in ApplicationController. In ApplicationController, we can realize before_actions to common behavior cross controllers)